src/js/pm-image-editor.js

```diff
@@ -0,0 +1,364 @@
+                this.wasCroppedForRotation = this.rotation % 2;
+            };
+
+            /**
+             * Made horizontal flip based on current image data.
+             */
+            ImageEditorFactory.prototype.horizontalFlip = function() {
+                this.hFlip = !this.hFlip;
+
+                var s = this.parentSize();
+                // Formula became from:
+                // 1. Moving center to point (0, y0 + h/2).
+                // 2. Miror move of Y axe coordinates.
+                // 3. Shift visible area back by Y axe.
+                // 4. Moving center to point (0, -(y0 + h/2)).
+                this.top = s.height - this.height - this.top; 
+            };
+
+            /**
+             * Made vertical flip based on current image data.
+             */
+            ImageEditorFactory.prototype.verticalFlip = function() {
+                this.vFlip = !this.vFlip;
+
+                var s = this.parentSize();
+                // Formula became from:
+                // 1. Moving center to point (x0 + w/2, 0).
+                // 2. Miror move of X axe coordinates.
+                // 3. Shift visible area back by X axe.
+                // 4. Moving center to point (-(x0 + w/2), 0).
+                this.left = s.width - this.width - this.left; 
+            };
+
+
+            /**
+             * Made image rotation flip based on current image data.
+             * Direction can be 'cw' - rotate by clock-wise direction
+             * and 'acw' - rotate by anti-clock-wise direction.
+             *
+             * @param 'cw'|'acw' direction - rotation direction.
+             */
+            ImageEditorFactory.prototype.rotate = function(direction) {
+                //this.isCropped = false;
+                // Update rotation value depends on direction.
+                this.rotation += direction === 'cw' ? 1 : -1;
+                // Make sure that rotation stays positive in range 0-3.
+                this.rotation = (this.rotation + 4)%4;
+
+                var s = this.parentSize();
+
+                // Update width and height:
+                var r = s.height/this.visibleWidth;
+                // Height depends on difference between visible width and parent height.
+                // Because this values are changing during rotation.
+                this.height = this.height*r;
+                // Keep proportions using image ratio.
+                this.width = this.height*this.ratio;
+
+                // Set of help variables.
+                // Top and left coordinates should be multiplyed on the same value as height,
+                // because it is scale transformation.
+                var y0 = this.top*r,
+                    x0 = this.left*r,
+                    w1 = s.width,
+                    h1 = s.height,
+                    w = this.width,
+                    h = this.height,
+                    a = x0 + w/2,
+                    b = y0 + h/2;
```

Comment:
This is shorteners for names again. I've used same names in formulas. Those vars have no sence actually, so cant give proper names for it.